### PR TITLE
Github: Fix bump-dependencies runs on existing PRs & adjust title

### PR DESCRIPTION
**Pull request by hoffie**
**Short description of changes**

The bump-dependencies workflow got permission errors from the Github API when trying to edit existing PRs which had already been added to an organization project.
The reason for this is that the GITHUB_TOKEN of the run is scoped to the repo, but `gh edit` tries to fetch all fields of a PR which includes the inaccessible organization project field. Therefore, use `gh api` instead which can be used in a more fine-grained
way.

This PR also addresses @ann0see's request to mimic the dependabot commit message style.

Group with:
CHANGELOG: Internal: Enabled automated dependency updates via dependabot and custom automation

**Context: Fixes an issue?**

Fixes: https://github.com/jamulussoftware/jamulus/pull/2777#issuecomment-1222793504
Fixes: https://github.com/jamulussoftware/jamulus/pull/2787#issuecomment-1222783064

**Does this change need documentation? What needs to be documented and how?**

No.

**Status of this Pull Request**

Ready.

**What is missing until this pull request can be merged?**
Reviews.

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

File: .github/workflows/bump-dependencies.yml [hunk 1]
```diff
@@ -137,7 +137,14 @@ jobs:
           fi
           if [[ $existing_pr ]]; then
             existing_title="$(gh pr view "${existing_pr}" --json title --jq .title)"
-            gh pr edit "${existing_pr}" --title "${pr_title}" --body "${body}"
+            # Use Github's API directly instead of using `gh edit`.
+            # The latter internally queries all fields of the PR which leads to
+            # a permission error as the workflow does not have access to
+            # organization projects:
+            gh api --silent -H "Accept: application/vnd.github+json" --method PATCH \
```
Comment (pljones) [inline .github/workflows/bump-dependencies.yml:144]: Is `--silent` a good idea?  I know the script will fail if `gh` returns non-zero - but will the log contain the error message with `--silent`?

Comment (hoffie) [reply .github/workflows/bump-dependencies.yml:144]: > but will the log contain the error message with --silent?

I think so.
Without `--silent`, the output will always contain the full JSON which sounds rather annoying.

Comment (pljones) [reply .github/workflows/bump-dependencies.yml:144]: If we want stderr but not stdout... seems to me we should silence stdout (`>&-`) rather than both.

Comment (hoffie) [reply .github/workflows/bump-dependencies.yml:144]: I've just run some tests:

**GET existing pull request without --silent**
```
$ gh api  -H "Accept: application/vnd.github+json" --method GET "/repos/jamulussoftware/jamulus/pulls/2801"
{
  "url": "https://api.github.com/repos/jamulussoftware/jamulus/pulls/2801",
  "id": 1033348737,
```

**GET existing pull request with --silent**
```
$ gh api --silent -H "Accept: application/vnd.github+json" --method GET "/repos/jamulussoftware/jamulus/pulls/2801"
$
```

**GET non-existing pull request with --silent**
```
$ gh api --silent -H "Accept: application/vnd.github+json" --method GET "/repos/jamulussoftware/jamulus/pulls/0"
gh: Not Found (HTTP 404)
```

**GET non-existing pull request without --silent**
```
$ gh api -H "Accept: application/vnd.github+json" --method GET "/repos/jamulussoftware/jamulus/pulls/0"
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest/reference/pulls#get-a-pull-request"
}
gh: Not Found (HTTP 404)
```

Conclusion: `gh` properly logs errors, even with `--silent`. I don't see any benefit to have the request body in addition, nor do I see a way to easily do that (the body is always on stdout).

Therefore, I think it makes most sense to keep using `--silent` without further adjustments.
@pljones What do you think?

Comment (pljones) [reply .github/workflows/bump-dependencies.yml:144]: OK, that's good enough.

**Review by pljones — APPROVED** [review]

**Review by ann0see — APPROVED** [review]

**Pull request merged**